go/vt/vtgate/vindexes/binarymd5.go

```diff
@@ -26,18 +26,23 @@ import (
 )
 
 var (
-	_ SingleColumn = (*BinaryMD5)(nil)
-	_ Hashing      = (*BinaryMD5)(nil)
+	_ SingleColumn    = (*BinaryMD5)(nil)
+	_ Hashing         = (*BinaryMD5)(nil)
+	_ ParamValidating = (*BinaryMD5)(nil)
 )
 
 // BinaryMD5 is a vindex that hashes binary bits to a keyspace id.
 type BinaryMD5 struct {
-	name string
+	name   string
+	params map[string]string
 }
 
-// NewBinaryMD5 creates a new BinaryMD5.
-func NewBinaryMD5(name string, _ map[string]string) (Vindex, error) {
-	return &BinaryMD5{name: name}, nil
+// newBinaryMD5 creates a new BinaryMD5.
+func newBinaryMD5(name string, params map[string]string) (Vindex, error) {
+	return &BinaryMD5{
+		name:   name,
+		params: params,
```

Comment:
Why not store the unknown params directly like it is done for `binary` vindex?

Reply:
Hey @GuptaManan100 the `binary_md5` vindex does not recognize any params, so any params that are passed in are unknown. So in this case it makes no difference from either a functional perspective or in terms of how memory gets used. But I see the argument for doing things the same way in all cases. I'll plan on making this more consistent in follow-up PR.

Reply:
Okay 👍